From ed50903e457430b30ebe4c9f65324bb38074a90a Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Mon, 30 Nov 2015 15:08:41 +0100
Subject: [PATCH 1/3] Add some optimization at the RAM initialisation

There are some optimizations in the RAM initialisation which depends on
the score hardware
---
 board/freescale/mx6sxscore/imximage.cfg | 28 +++++++++++++++++-----------
 1 file changed, 17 insertions(+), 11 deletions(-)

diff --git a/board/freescale/mx6sxscore/imximage.cfg b/board/freescale/mx6sxscore/imximage.cfg
index bdb5b95..8f699ab 100644
--- a/board/freescale/mx6sxscore/imximage.cfg
+++ b/board/freescale/mx6sxscore/imximage.cfg
@@ -56,34 +56,40 @@ CSF 0x2000
 /* Enable all clocks */
 DATA 4 0x020c4068 0xffffffff
 DATA 4 0x020c406c 0xffffffff
-DATA 4 0x020c4070 0xffffffff
+//DATA 4 0x020c4070 0xffffffff
+DATA 4 0x020c4070 0xfc3fffff
 DATA 4 0x020c4074 0xffffffff
 DATA 4 0x020c4078 0xffffffff
 DATA 4 0x020c407c 0xffffffff
 DATA 4 0x020c4080 0xffffffff
-DATA 4 0x020c4084 0xffffffff
+//DATA 4 0x020c4084 0xffffffff
 
 /* IOMUX */
 /* DDR IO TYPE */
+// Select DDR3 Mode
 DATA 4 0x020e0618 0x000c0000
+// Disable Pull/Keeper for RAM
 DATA 4 0x020e05fc 0x00000000
 
-// TODO: Maybe a problem
 /* CLOCK */
 // CLK0 Drive Strength 40 Ohm
 DATA 4 0x020e032c 0x00000030
 
 /* ADDRESS */
 // CAS Drive Strength 60 Ohm
-DATA 4 0x020e0300 0x00000020
+//DATA 4 0x020e0300 0x00000020
+// Don't touch Pull-Up Pull-Down Configuration
+DATA 4 0x020e0300 0x00008020
 // RAS Drive Strenth 60 Ohm
-DATA 4 0x020e02fc 0x00000020
+//DATA 4 0x020e02fc 0x00000020
+DATA 4 0x020e02fc 0x00008020
 // ADDR Drive Strength 60 Ohm
 DATA 4 0x020e05f4 0x00000020
 
 /* CONTROL */
 // DRAM Reset Drive Strength 60 Ohm
-DATA 4 0x020e0340 0x00000020
+//DATA 4 0x020e0340 0x00000020
+DATA 4 0x020e0340 0x000C0020
 // SDBA2
 DATA 4 0x020e0320 0x00000000
 // ODT0 Drive Strength 60 Ohm
@@ -106,9 +112,9 @@ DATA 4 0x020e0338 0x00000028
 DATA 4 0x020e033c 0x00000028
 
 /* DATA */
-
-// Data lines Differential
+// Data lines CMOS
 DATA 4 0x020e0608 0x00020000
+//DATA 4 0x020e0608 0x00000000
 // DATA0-7  Drive Strength 48 Ohm
 DATA 4 0x020e060c 0x00000028
 // DATA8-15 Drive Strength 48 Ohm
@@ -123,7 +129,7 @@ DATA 4 0x020e02ec 0x00000028
 DATA 4 0x020e02f0 0x00000028
 // DQM2 Drive Strength 48 Ohm
 DATA 4 0x020e02f4 0x00000028
-// DQM2 Drive Strength 48 Ohm
+// DQM3 Drive Strength 48 Ohm
 DATA 4 0x020e02f8 0x00000028
 
 /* Calibrations */
@@ -152,7 +158,7 @@ DATA 4 0x021b0828 0x33333333
 DATA 4 0x021b08b8 0x00000800
 
 /* MMDC init */
-/* in DDR3, 64-bit mode, only MMDC0 is initiated */
+/* in DDR3, 32-bit mode, only MMDC0 is initiated */
 DATA 4 0x021b0004 0x0002002d
 // ODT Timing
 DATA 4 0x021b0008 0x00333030
@@ -171,7 +177,7 @@ DATA 4 0x021b002c 0x000026d2
 // Out of Reset Delays
 DATA 4 0x021b0030 0x006b1023
 // Address Space Partition
-DATA 4 0x021b0040 0x0000005f
+DATA 4 0x021b0040 0x0000004f
 // Core Control Register
 // 13 Bits Row
 // 10 Bits Col
-- 
1.9.1

